app/views/devise/registrations/edit.html.erb

```diff
@@ -1,43 +1,69 @@
+    <% if resource.errors.any? %>
+      <div class="panel alert">
+        <%= devise_error_messages! %>
+      </div>
+    <% end %>
 
-  <div class="field">
-    <%= f.label :email %><br />
-    <%= f.email_field :email, autofocus: true %>
-  </div>
+    <div class="field">
+      <%= f.label :username %>
+      <%= f.text_field :username,
+      autofocus: true,
+      placeholder: 'Username'
+      %>
+    </div>
+
+    <div class="field">
+      <%= f.label :email %>
+      <%= f.email_field :email,
+      placeholder: 'Email'
+      %>
+    </div>
 
-  <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
+    <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
     <div>Currently waiting confirmation for: <%= resource.unconfirmed_email %></div>
-  <% end %>
-
-  <div class="field">
-    <%= f.label :password %> <i>(leave blank if you don't want to change it)</i><br />
-    <%= f.password_field :password, autocomplete: "off" %>
-    <% if @minimum_password_length %>
-      <br />
-      <em><%= @minimum_password_length %> characters minimum</em>
     <% end %>
-  </div>
 
-  <div class="field">
-    <%= f.label :password_confirmation %><br />
-    <%= f.password_field :password_confirmation, autocomplete: "off" %>
-  </div>
 
-  <div class="field">
-    <%= f.label :current_password %> <i>(we need your current password to confirm your changes)</i><br />
-    <%= f.password_field :current_password, autocomplete: "off" %>
-  </div>
+    <div class="field">
+      <%= f.label :password, class: 'left' %>
+      <i class='right'>(leave blank if you don't want to change it)</i>
+      <%= f.password_field :password,
+      autocomplete: "off",
+      placeholder: 'Password'
+      %>
+    </div>
 
-  <div class="actions">
-    <%= f.submit "Update" %>
-  </div>
-<% end %>
+    <div class="field">
+      <%= f.label :password_confirmation, 'Confirm Password' %>
+      <%= f.password_field :password_confirmation,
+      autocomplete: "off" ,
+      placeholder: 'Confirm Password'
+      %>
+    </div>
+
+    <div class="field">
+      <%= f.label :current_password, 'Current Password', class: 'left'%>
+      <i class='right'>(we need your current password to confirm your changes)</i>
+      <%= f.password_field :current_password,
+      autocomplete: "off",
+      placeholder: 'Current Password'
+      %>
+    </div>
 
-<h3>Cancel my account</h3>
+    <div class="actions">
+      <%= f.submit "Update", class: 'button' %>
+    </div>
+    <% end %>
 
-<p>Unhappy? <%= button_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?" }, method: :delete %></p>
+    <h3>Cancel my account</h3>
 
-<%= link_to "Back", :back %>
+    <p>Unhappy? <%= button_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?" }, method: :delete, class: 'button alert' %></p>
```

Comment:
Looks like this is going to say "Cancel my account" twice -- is this definitely what you want?